code/game/objects/structures/survey_probe.dm

```diff
@@ -0,0 +1,125 @@
+					SPAN_NOTICE("You fasten the locking bolts \the [src], stowing it. It retracts its devices and drill bits."),
+					)
+				item.play_tool_sound(user, 30)
+				anchored = FALSE
+				icon_state = "surveying_probe"
+
+/obj/structure/survey_probe/attack_hand(mob/user as mob)
+	if(!timer_id)
+		if(anchored)
+			user.visible_message(
+				SPAN_NOTICE("\The [user] activates \the [src], starting the surveying process."),
+				SPAN_NOTICE("You activate \the [src], starting the surveying process. It starts drilling and sampling the ground and air."),
+				)
+			timer_id = addtimer(CALLBACK(src, PROC_REF(survey_end)), 20 SECONDS)
+			icon_state = "surveying_probe_active"
+		else
+			to_chat(user, SPAN_NOTICE("You try to activate \the [src], but its devices and drill bits are not deployed yet."))
+
+/obj/structure/survey_probe/proc/survey_end()
+	if(timer_id)
+		// message
+		src.visible_message(
+			SPAN_NOTICE("\The [src] finishes the surveying process, and prints out a report."),
+			)
+
+		// report vars and default vals
+		var/report_location = "unknown/invalid"
+		var/ground_report = "Invalid or insufficient data, ground survey unsuccessful. "
+		var/atmos_report = "Invalid or insufficient data, atmospheric survey unsuccessful. "
+
+		// turf
+		var/turf/turf = src.loc
+		var/turf_is_exoplanet = istype(turf, /turf/simulated/floor/exoplanet)
+		var/turf_is_hard_floor = istype(turf, /turf/simulated/floor/tiled)
+		var/turf_is_fake_grass = istype(turf, /turf/simulated/floor/grass)
+
+		// atmos
+		var/datum/gas_mixture/air = turf.return_air()
+		if(air.total_moles>0)
+			atmos_report = english_list(atmosanalyzer_scan(turf, air))
+
+		// not exoplanet turf
+		if(turf_is_hard_floor)
+			ground_report += "The probe cannot penetrate the hard metal floor."
+		else if(turf_is_fake_grass)
+			ground_report += "The probe detects soft soil, but it cannot penetrate the ground deep enough to get any meaningful data."
+		else if(!turf_is_exoplanet)
+			ground_report += "The probe cannot penetrate the ground deep enough to get any meaningful data."
+
+		// ground survey from sector / exoplanet
+		if(turf_is_exoplanet && current_map.use_overmap)
+			var/obj/effect/overmap/visitable/sector/sector = GLOB.map_sectors["[z]"]
+			var/obj/effect/overmap/visitable/sector/exoplanet/exoplanet = sector
+			if(istype(sector))
+				report_location = sector.name
+				ground_report = ""
+				if(istype(exoplanet))
+					ground_report += "<br><b>Estimated Mass and Volume: </b>[exoplanet.massvolume]BSS(Biesels)"
+					ground_report += "<br><b>Surface Gravity: </b>[exoplanet.surfacegravity]Gs"
+					ground_report += "<br><b>Geological Variables: </b>[exoplanet.geology]"
+					ground_report += "<br><b>Surface Water Coverage: </b>[exoplanet.surfacewater]"
+					ground_report += "<br><b>Apparent Weather Data: </b>[exoplanet.weather]"
+					ground_report += "<br>"
+				if(sector.ground_survey_result)
+					ground_report += sector.ground_survey_result
```

Comment:
Shift this around so the sector specific things happen after the relevant check, then the exoplanet-specific things happen after the relevant check

Reply:
done I think

Reply:
the exoplanet stuff should be first so it appears first in the paper though

Reply:
```
var/exoplanet_stuff = ""
if(exoplanet_stuff_whatever)
	exoplanet_stuff += "something"

var/ground_stuff = ""
if(ground_stuff_whatever)
	ground_stuff += "something"

report += exoplanet_stuff
report += ground_stuff
```
It would have to be more like this, and I think that does not look better than it does currently.